tests/kafka_cluster_manager/rg_test.py

```diff
@@ -615,3 +615,30 @@ def test_rebalance_brokers_for_topic_partition_imbalance(self, create_partition)
             b1.topics == possible_topics2 and
             b2.topics == possible_topics1
         )
+
+    def test_add_replica(self, create_partition):
+        p10 = create_partition('topic1', 0)
+        p20 = create_partition('topic2', 0)
+        b1 = create_broker('b1', [p10, p20])
+        b2 = create_broker('b2', [p10, p20])
+        b3 = create_broker('b3', [p20])
+        rg = ReplicationGroup('test_rg', set([b1, b2, b3]))
+        rg.add_replica(p10)
+
+        assert b1.partitions == set([p10, p20])
+        assert b2.partitions == set([p10, p20])
+        assert b3.partitions == set([p10, p20])
+
+    def test_remove_replica(self, create_partition):
```

Comment:
Add a test where the condition broker in rg is not verified (exception or assertion is thrown).